distribution/openrct2.d.ts

```diff
@@ -122,6 +122,7 @@ declare global {
      */
     interface CarTrackLocation extends CoordsXYZD {
         trackType: number;
+        onCoveredTrack: boolean;
```

Comment:
Does this also fit future plans where more/other alternative track options may become available? Ideally we'd minimize the amount of changes if more changes are planned for transitioning to more flexibility.

Reply:
Second this, wouldn't:
```suggestion
        onAlternateTrack: boolean;
```
or
```suggestion
        onSecondaryTrack: boolean;
```
be better/more future proof options? Covered is already a strange way to differentiate in the context of the water coaster.

Reply:
There is a better way than a bool that has been devised but hasn't been implemented into this PR.

Reply:
> Covered is already a strange way to differentiate in the context of the water coaster.

The water coaster would not get the covered flag, as I understand. The covered flag only controls whether the train can fly off the track, and for the water coaster, I don't think the trains can fly off the track even on water segments.

Reply:
This is on the plugin interface. Water coaster uses the covered track piece IDs for water segments.